File: solutions/py/6 kyu/WhoLikesIt.py
#!/usr/bin/python3

# https://www.codewars.com/kata/5266876b8f4bf2da9b000362/
def likes(names):
    return f"{('no one' if not len(names) else names[0])}{'' if len(names) < 2 else f' and {names[1]}' if len(names) == 2 else f', {names[1]} and {names[2]}' if len(names) == 3 else f', {names[1]} and {len(names) - 2} others'} {'likes' if len(names) <=1 else 'like'} this"

File: solutions/scala/6 kyu/WhoLikesIt.scala
//https://www.codewars.com/kata/5266876b8f4bf2da9b000362/
object Sol {
  def likes(names: Array[String]): String = {
    names.length match
    case 0 => "no one likes this"
    case 1 => "%s likes this".format(names(0))
    case 2 => "%s and %s like this".format(names: _*)
    case 3 => "%s, %s and %s like this".format(names: _*)
    case _ => "%s, %s and %s others like this".format(names(0), names(1), names.length - 2)
  }
}